src/model/train.py

```diff
@@ -1,19 +1,29 @@
+import cv2
+import mlflow
 import torch
+from sklearn.metrics import accuracy_score, classification_report, f1_score, precision_score, recall_score
 from tqdm import tqdm
 
 from src.preprocessing.dataloader import Dataloader
 
 
 class Model:
-    def __init__(self, model: torch.nn.Module, device: torch.device, data_loader: Dataloader) -> None:
+    def __init__(
+        self,
+        model: torch.nn.Module,
+        data_loader: Dataloader,
+        optimizer: torch,
+        loss_fn: torch = torch.nn.CrossEntropyLoss(),
+        device: torch.device = None,
```

Comment:
`mypy` should give you an error in this line.
```suggestion
        device: torch.device | None = None,
```